### PR TITLE
Working branch

### PR DESCRIPTION
**Pull request by JelilFaisalAbudu**
# Requesting Code Review for Association(Private Events) Project

* I have finished building the project and it's now ready for your review

### Sign-up and Sign-in Processes
* User signs up with his username and email - all required
* Simple log-in process with email address(by using sessions to store user id)

### Interactions with the Webapp
* User can create an event for other to attend
* User can register to attend several events  in case the attendance date hasn't passed

## Unit Testing

* I've performed the unit test for the model(Event, User and Invitation) associations
* Also performed integration test user authentication and event management

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,12 +1,14 @@
 Rails.application.routes.draw do
+  resources :events, only: [:new, :index, :show, :create, :attended_event, :unattend_event]
```
Comment (Dandush03) [inline config/routes.rb:2]: __OPTIONAL__: You could use the `except` flag instead of the `only` flag

File: config/routes.rb [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
   get 'events/show'
-  resources :users
-  resources :events
-  
+  post 'add_events_attending', to: 'events#attended_event'
+  post 'rem_events_attending', to: 'events#unattend_event'
+  get '/signup', to: 'users#new'
   get '/login', to: 'sessions#new'
   post '/login', to: 'sessions#create'
   delete '/logout', to: 'sessions#destroy'
```
Comment (Dandush03) [inline config/routes.rb:12]: __OPTIONAL__: You could use resource check out [this link](https://guides.rubyonrails.org/routing.html)

**Review by Dandush03 — CHANGES_REQUESTED** [review]
# Status: :red_circle: Require Change :red_circle: 
### Great work so far
__IMPORTANT__ before digging in into your code, I most say that your spec is failing when it tries to create a new event when a user is logged in, Apparently you added a field name `location` which is causing the issue.
### Now lest us dive in to your code :wink: 
# Happy Coding :computer: :nerd_face:

**Comment by JelilFaisalAbudu** [comment]
# Required Changes have been Made
* There is an unpermitted field which causes the test to fail. It has been resolved and now all specs are passing

**Review by rOluochKe — APPROVED** [review]
Hello @JelilFaisalAbudu 
### You got this right :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next one.
- Happy hacking!

**Pull request merged**